presto-main/src/main/java/io/prestosql/sql/planner/optimizations/HashGenerationOptimizer.java

```diff
@@ -839,7 +839,7 @@ public HashComputationSet withHashComputation(Optional<HashComputation> hashComp
             return Optional.empty();
         }
 
-        Expression result = new LongLiteral(String.valueOf(INITIAL_HASH_VALUE));
+        Expression result = new GenericLiteral(StandardTypes.BIGINT, String.valueOf(INITIAL_HASH_VALUE));
```

Comment:
Would it fail somehow? Is it possible to write test?

Reply:
It doesn't fail because it's used only in a benchmark that relies on the interpreter. The interpreter "compensates" for this by doing the implicit cast. 

It does fail when trying to compile the expression, which is what I did in a follow commit to remove the interpreted processors.